interfaces/sourcegen/sourcegen/_data/ctsol_auto.yaml

```diff
@@ -0,0 +1,43 @@
+# Auto-generated CLib API for %Cantera's Solution class.
+# Partially implements a replacement for CLib's traditional @c ct library.
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+prefix: sol3
+base: Solution
+parents: []  # List of parent classes
+derived: [Interface]  # List of specializations
+recipes:
+- name: newSolution
+  implements: newSolution(const string&, const string&, const string&)
+  uses: [thermo, kinetics, transport]
+- name: newInterface
+  implements:
+    newInterface(const string&, const string&, const vector<shared_ptr<Solution>>&)
+  uses: [thermo, kinetics]
+- name: del
+  uses: [thermo, kinetics, transport]
+- name: name
+- name: setName
+- name: thermo
+- name: kinetics
+- name: transport
+- name: setTransportModel
+  code: |-
+    try {
+        auto obj = SolutionCabinet::at(handle);
+        TransportCabinet::del(
+            TransportCabinet::index(*(obj->transport()), handle));
+        obj->setTransportModel(model);
+        return TransportCabinet::add(obj->transport(), handle);
+    } catch (...) {
+        return handleAllExceptions(-2, ERR);
+    }
```

Comment:
On second thought, the custom code section I added here should be moved from this main configuration file to `clib/config.yaml` as it is an implementation detail. This isn't difficult to change, but I'll hold off until there are other review comments.

Reply:
It’s been a while since this all posted. As the PR is approved, I’ll merge and address this in a follow up.

Reply:
@speth ... I saw your comment and will hold off on the merge.

Reply:
As mentioned above, I will move the 'custom code' portion in a future PR.